### PR TITLE
Graph Dictionary pattern

### PR DESCRIPTION
**Pull request by darrelmiller**
Here's a first effort at converting the existing Dictionary documentation to the pattern format.  I also created a patterns folder for all pattern related docs to live.

### CODE REVIEW

File: graph/patterns/dictionary/index.md [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
+--------
+
+The API design requires a resource to include an unknown quantity of data elements of the same type that must be named using values provided by the API consumer.
+
+## Solution
+--------
+
+API designers use a JSON object to represent a dictionary in a `application/json`response payload. When describing the model in CSDL, a new complex type can be created that derives from `Org.OData.Core.V1.Dictionary` and then use the `Org.OData.Validation.V1.OpenPropertyTypeConstraint`to constrain the type that can be used for the values in the dictionary.
+
+Dictionary entries can be added via `POST`, updated via `PATCH`, and they can be removed by setting the entry value to `null`. Multiple dictionaries can be updated at once by using `PATCH` on the dictionary property.
```
Comment (gareth-a-jones) [inline graph/patterns/dictionary/index.md]: Multiple dictionaries?   Multiple entries?

Comment (gareth-a-jones) [inline graph/patterns/dictionary/index.md]: Surprised they support POST.  I would expect only PATCH.

File: graph/patterns/dictionary/index.md [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
 More information:
 
 - [OData reference](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Core.V1.md#dictionary)
 
-## When to use dictionary types
+ 
+## When to Use this Pattern
+------------------------
 
 Before using a dictionary type in your API definition make sure your scenario fits the following criteria:
```
Comment (gareth-a-jones) [inline graph/patterns/dictionary/index.md]: "The client MUST define the keys of this type. As opposed to the service defining it in advance."
"The client MUST define the keys of this type. As opposed to the service defining them in advance.

File: graph/patterns/dictionary/index.md [hunk 3]
```diff
@@ -147,13 +178,13 @@ PATCH https://graph.microsoft.com/v1.0/users/10/roles/author
...
 
-### Deleting an entry from the dictionary
+#### Deleting an entry from the dictionary
 
 ```HTTP
 DELETE https://graph.microsoft.com/v1.0/users/10/roles/author
 ```
 
-## CDSL example
+### CDSL example
```
Comment (gareth-a-jones) [inline graph/patterns/dictionary/index.md]: ""

No idea what this is about but it needs more explanation.

File: graph/patterns/dictionary/index.md (hunk 1 above)
Comment (darrelmiller) [reply graph/patterns/dictionary/index.md]: Should be multiple entries.
```suggestion
Dictionary entries can be added via `POST`, updated via `PATCH`, and they can be removed by setting the entry value to `null`. Multiple entries can be updated at once by using `PATCH` on the dictionary property.
```

Comment (darrelmiller) [reply graph/patterns/dictionary/index.md]: Yeah me too.  That's why I have an open question later on in the doc.

File: graph/patterns/dictionary/index.md [hunk 4]
```diff
@@ -1,12 +1,40 @@
...
+
+## Issues and Considerations
+-------------------------
 
 Dictionaries, sometimes called maps, are a collection of name-value pairs. They allow dynamic data sets to be accessed in a systematic manner and are a good compromise between a strictly defined ahead of time structure with all its named properties and between a loosely defined dynamic object (i.e. OData OpenTypes).
 
+As dictionary entries are removed via setting the value to null, this means that dictionaries can only support values that are non-nullable.
+
+  OpenQuestions:
+  -  Can/should PUT be supported on the dictionary property and/or the entry value
```
Comment (baywet) [inline graph/patterns/dictionary/index.md]:
```suggestion
  - Can/should PUT be supported on the dictionary property and/or the entry value 
```

**Review by baywet — APPROVED** [review]

File: graph/patterns/dictionary/index.md (hunk 1 above)
Comment (OlgaPodo) [reply graph/patterns/dictionary/index.md]: Just verifying if a dictionary can be created via POST?

**Pull request merged**